sktime/forecasting/tests/test_all_forecasters.py

```diff
@@ -554,10 +554,16 @@ def test_predict_proba(self, estimator_instance, n_columns, fh_int_oos):
                 and no NotImplementedError is raised when calling predict_proba
         """
         y_train = _make_series(n_columns=n_columns)
-        estimator_instance.fit(y_train, fh=fh_int_oos)
 
         if estimator_instance.get_tag("capability:pred_int"):
             try:
+                # y_train.name is "None" (str)
+                estimator_instance.fit(y_train, fh=fh_int_oos)
+                pred_dist = estimator_instance.predict_proba()
+                self._check_predict_proba(pred_dist, y_train, fh_int_oos)
+                # y_train.name is None (None)
+                estimator_instance.fit(y_train, fh=fh_int_oos)
+                y_train.name = None
```

Comment:
are these two lines not in the wrong order? I.e., set the `name` and then pass it to `fit`?

Reply:
Will modify, test and push the changes.

Reply:
The data generated by `_make_data(n_columns=n_columns)` has name `None` of type `None` initially (`y_train.name`).
For some reason, the `fit` method is causing this:
```
Before fit:  None <class 'NoneType'>
After fit:  None <class 'str'>
```
Code to reproduce.
```
# Replace line 565 with following three lines in test_all_forecasters.py of this PR.

print("Before fit: ", y_train.name, type(y_train.name))
estimator_instance.fit(y_train, fh=fh_int_oos)
print("After fit: ", y_train.name, type(y_train.name))
```

Reply:
of which estimator, `ARCH`?
Do you know if this happens with all estimators, or only some?

What's strange is that we have tests for no side effects, these should catch `y_train` being mutated.

More precisely, for all scenarios (data) in `utils._testing.scenarios_forecasting` (that have the "enabled" tag), we test that none of the methods mutate the data. So it might be interesting to understand why it's not being caught.

Reply:
Ohkay, I will take a closer look at it.

Reply:
This isn't the case with any estimator other than `ARCH`. This effect was caused by `arch` (`arch_model`). I fixed it by passing deep copies of `X`, `y` to 'arch_model' rather than just `X`, `y`. Kindly review the code.